applications/CompressiblePotentialFlowApplication/custom_conditions/potential_wall_condition.h

```diff
@@ -202,6 +206,125 @@ class PotentialWallCondition : public Condition
         return pNewCondition;
     }
 
+    /// Find the condition's parent element.
+	void Initialize() override
+	{
+        KRATOS_TRY;
+
+        if(this->Id()==1)
+        {
+            std::cout << "Condition = " << this->Id()  << std::endl;
+            std::cout << "mInitializeWasPerformed = " << mInitializeWasPerformed  << std::endl;
+        }
+        
+        //std::cout << "Condition = " << this->Id()  << std::endl;
+
+		const array_1d<double,3>& rNormal = this->GetValue(NORMAL);
+		if (norm_2(rNormal) == 0.0)
+		  {
+		    std::cout << "error on condition -> " << this->Id() << std::endl;
+		    KRATOS_THROW_ERROR(std::logic_error, "NORMAL must be calculated before using this condition","");
+		  }
+
+		if (mInitializeWasPerformed)
+		{
+            std::cout << "Returning early"  << std::endl;
+            std::cout << "Condition = " << this->Id()  << std::endl;
+			return;
+		}
+
+		mInitializeWasPerformed = true;
+
+		double EdgeLength;
+		array_1d<double,3> Edge;
+		GeometryType& rGeom = this->GetGeometry();
+		WeakPointerVector<Element> ElementCandidates;
+		for (SizeType i = 0; i < TDim; i++)
```

Comment:
It will be cleaner if you split up this long function into subroutines. This code block could be called here as `GetElementCandidates(...)`. Also, this code looks like it will not remove duplicate candidates from the vector. Shouldn't each candidate be unique?

Reply:
Thanks for the hint. I will split it up into subroutines. I believe it is not necessary to remove duplicate candidates. However I did not implemente this function myself, I am using the implementation from Riccardo and Poyaan that can be found in [here](https://github.com/KratosMultiphysics/Kratos/blob/master/applications/FluidDynamicsApplication/custom_conditions/fs_generalized_wall_condition.h). I am interested in discussing this.

Reply:
It is my fault if it doesn't do the right thing since I wrote that function. We can discuss it.